packages/rich-text/src/plugins/Tracking/createTrackingPlugin.ts

```diff
@@ -111,5 +111,43 @@ export const createTrackingPlugin = (onAction: RichTextTrackingActionHandler): R
 
       return editor;
     },
+    then: (editor) => ({
+      editor: {
+        insertData: {
+          format: 'text/html',
+          getFragment: ({ data }) => {
+            let source;
+
+            if (data.includes('docs-internal-guid')) {
+              source = 'Google Docs';
+            }
+
+            if (data.includes('<google-sheets-html-origin>') || data.includes('data-sheets-')) {
+              source = 'Google Spreadsheets';
+            }
+
+            if (data.includes('class="TextRun') || data.includes('class="OutlineElement')) {
```

Comment:
This feels too generic. Is the only way we can tell it's MS Word Online? Also what about normal MS Word and Excel Sheets?